src/sentry/utils/batching_kafka_consumer.py

```diff
@@ -177,18 +216,33 @@ def __init__(
         if queued_max_messages_kbytes is None:
             queued_max_messages_kbytes = DEFAULT_QUEUED_MAX_MESSAGE_KBYTES
 
-        self.consumer = self.create_consumer(
-            topics,
-            cluster_name,
-            group_id,
-            auto_offset_reset,
-            queued_max_messages_kbytes,
-            queued_min_messages,
-        )
-
         self.producer = producer
         self.commit_log_topic = commit_log_topic
         self.dead_letter_topic = dead_letter_topic
+        if not consumer:
+            self.consumer = self.create_consumer(
+                topics,
+                cluster_name,
+                group_id,
+                auto_offset_reset,
+                queued_max_messages_kbytes,
+                queued_min_messages,
+            )
+        else:
+            assert isinstance(consumer, KafkaConsumerFacade)
```

Comment:
Please set the type hint for the consumer parameter, so you do not need the isinstance.

Reply:
This file is not in mypy.ini. That's why I added the runtime check here